changelog/31255.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:change
+secrets/artifactory: Use Artifactory identity token for auth, instead of username with API key
```

Comment:
Since we're only changing the behavior of Vault CI, not the Vault binary itself, we don't need a changelog. Instead, use the `pr/no-changelog` label to get rid of failed check.